core/drivers/clk/sam/at91_generated.c

```diff
@@ -72,16 +73,27 @@ static TEE_Result clk_generated_set_parent(struct clk *clk, size_t index)
 	if (index >= clk_get_num_parents(clk))
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	gck->parent_id = index;
+	if (gck->mux_table)
+		gck->parent_id = gck->mux_table[index];
+	else
+		gck->parent_id = index;
 
 	return TEE_SUCCESS;
 }
 
 static size_t clk_generated_get_parent(struct clk *clk)
 {
 	struct clk_generated *gck = clk->priv;
-
-	return gck->parent_id;
+	unsigned int i = 0;
+
+	if (gck->mux_table) {
+		for (i = 0; i < gck->mux_table_size; i++)
```

Comment:
nitpicking: `clk_generated_set_parent()` relies on `index >= clk_get_num_parents(clk)` and here on `gck->mux_table_size`. For consistency, I think you could remove field `mux_table_size` from `struct clk_generated` and use `clk_get_num_parents(clk)` here also (or use straight `clk->num_parents` as done in `clk_sama7g5_master_xxx()`).